src/containers/LoginRegister/index.jsx

```diff
@@ -0,0 +1,224 @@
+import React, { useContext, useState } from "react";
+import Login from "../../components/Login";
+import Register from "../../components/Register";
+import PasswordReset from "../../components/PasswordReset";
+import firestore, { auth } from "../../firebaseConfig";
+import firebase from "firebase";
+import { AuthContext } from "../../auth/authContext";
+import { withRouter } from "react-router-dom";
+import { Alert, Spin } from "antd";
+import "./index.css";
+import { useTranslation } from "react-i18next";
+import kids from "../../images/LoginKids.svg";
+
+const LoginRegisterPage = (props) => {
+  const [t] = useTranslation();
+  const login = async (email, password) => {
+    setError("");
+    setMessage(<></>);
+    try {
+      setLoading(true);
+      await auth.signInWithEmailAndPassword(email, password);
+      setLoading(false);
+      props.history.push("/");
+    } catch (e) {
+      setLoading(false);
+      setError(e.message);
+    }
+  };
+  const passwordReset = async (email) => {
+    setError("");
+    setMessage(<></>);
+    try {
+      setLoading(true);
+      await auth.sendPasswordResetEmail(email);
+      setLoading(false);
+      setMessage(
+        <Alert
+          style={{ marginBottom: 10 }}
+          showIcon
+          message={
+            <div>
+              {t("loginRegister.anEmailsent")}
+              <b>{email}</b>
+              {t("loginRegister.anEmailsentcontain")}
+            </div>
+          }
+          type="success"
+        ></Alert>
+      );
+      setActive("login");
+    } catch (e) {
+      setLoading(false);
+      setError(e.message);
+    }
+  };
+  // Login with FACEBOOK
+  const loginWithFacebook = async () => {
+    setError("");
+    setMessage(<></>);
+    try {
+      setLoading(true);
+      let provider = new firebase.auth.FacebookAuthProvider();
+      provider.setCustomParameters({
+        display: "popup",
+      });
+      let u = await auth.signInWithPopup(provider);
+      console.log("facebook user: ", u.user);
+      setLoading(false);
+      props.history.push("/");
+    } catch (e) {
+      setLoading(false);
+      setError(e.message);
+    }
+  };
+  // Loogin with GOOGLE
+  const loginWithGoogle = async () => {
+    setError("");
+    setMessage(<></>);
+    try {
+      setLoading(true);
+      let provider = new firebase.auth.GoogleAuthProvider();
+      provider.addScope("https://www.googleapis.com/auth/contacts.readonly");
+      let u = await auth.signInWithPopup(provider);
+      console.log("google user: ", u.user);
```

Comment:
console logs. also it feels like this code was just copy pasted, can you make a function and make it so that there's no copy pasted code?